pkg/spec/merge.go

```diff
@@ -338,79 +312,108 @@ func mergeProperties(properties, properties2 spec.SchemaProperties, path *field.
 	return retProperties, retConflicts
 }
 
-func mergeResponses(responses, responses2 *spec.Responses, path *field.Path) (*spec.Responses, []conflict) {
-	if r, shouldReturn := shouldReturnIfNil(responses, responses2); shouldReturn {
-		return r.(*spec.Responses), nil
+func mergeResponses(responses, responses2 spec.Responses, path *field.Path) (spec.Responses, []conflict) {
+	if r, shouldReturn := shouldReturnIfEmptyResponses(responses, responses2); shouldReturn {
+		return r, nil
 	}
 
 	var retConflicts []conflict
 
-	retResponses := &spec.Responses{
-		ResponsesProps: spec.ResponsesProps{
-			Default:             responses.Default, // default will be the same because we are generating it
-			StatusCodeResponses: make(map[int]spec.Response),
-		},
-	}
-
-	statusCodeResponses := responses.StatusCodeResponses
-	statusCodeResponses2 := responses2.StatusCodeResponses
+	retResponses := spec.NewResponses()
 
 	// go over first responses list
 	// 1. merge mutual response code responses
-	// 2. add non mutual response code responses
-	for code, response := range statusCodeResponses {
-		if response2, ok := statusCodeResponses2[code]; ok {
-			mergedResponse, conflicts := mergeResponse(response, response2, path.Child(strconv.Itoa(code)))
+	// 2. add non-mutual response code responses
+	for code, response := range responses {
+		if response2, ok := responses2[code]; ok {
+			mergedResponse, conflicts := mergeResponse(response.Value, response2.Value, path.Child(code))
 			retConflicts = append(retConflicts, conflicts...)
-			retResponses.StatusCodeResponses[code] = *mergedResponse
+			retResponses[code] = &spec.ResponseRef{Value: mergedResponse}
 		} else {
-			retResponses.StatusCodeResponses[code] = statusCodeResponses[code]
+			retResponses[code] = responses[code]
 		}
 	}
 
-	// add non mutual parameters from the second list
-	for code := range statusCodeResponses2 {
-		if _, ok := statusCodeResponses[code]; !ok {
-			retResponses.StatusCodeResponses[code] = statusCodeResponses2[code]
+	// add non-mutual parameters from the second list
+	for code := range responses2 {
+		if _, ok := responses[code]; !ok {
+			retResponses[code] = responses2[code]
 		}
 	}
 
 	return retResponses, retConflicts
 }
 
-func mergeResponse(response, response2 spec.Response, path *field.Path) (*spec.Response, []conflict) {
+func mergeResponse(response, response2 *spec.Response, path *field.Path) (*spec.Response, []conflict) {
 	var retConflicts []conflict
 	retResponse := spec.NewResponse()
+	if response.Description != nil {
+		retResponse.WithDescription(*response.Description)
```

Comment:
maybe it's a better practice to do ret = ret.WithXXX() everywhere here (not to assume inner implementation)?

Reply:
> everywhere here (not to assume inner implementation)
not sure I understand

Reply:
The convention when using `WithXXX` is to set it inside and return the object, not sure why we need to do `ret = ret.WithXXX()` everywhere